packages/owl-bot/src/copy-code.ts

```diff
@@ -227,23 +294,114 @@ ${copyTagLine}`,
     };
     return result;
   }
-  const {sourceCommitHash, commitMsgPath} = await copyCode(
+}
+
+/**
+ * Replace characters that cannot be in a branch name with an underscore.
+ *
+ * More conservative than perfect implementation described here:
+ * https://stackoverflow.com/questions/3651860/which-characters-are-illegal-within-a-branch-name
+ * but I don't expect name collisions anyway so conservative is fine.
+ */
+export function toSafeBranchName(s: string): string {
+  return s.replace(/[^A-Za-z0-9-]/g, '_');
+}
+
+/**
+ * Converts a yaml path like '/Speech/.OwlBot.yaml' to a branch name
+ * like 'owl-bot-copy-Speech', into which we'll copy the code.
+ */
+export function branchNameForCopy(yamlPath: string): string {
```

Comment:
In order to avoid potential merge conflicts in the future, what if we appended a serialized number at the end of the branch? Perhaps owl-bot could save/look up the last branch number, and that way know which should be the next one?

Reply:
In what scenario would there be a potential merge conflict?

Reply:
If someone makes changes to an owl-bot branch either unknowingly, or thinking they're fixing a PR. But perhaps this is an unlikely scenario

Reply:
There won't be a merge conflict because Owl Bot will check out the owl-bot branch *with the user's changes*.

On line 197:
```ts
    cmd(`git clone --single-branch -b ${cloneBranch} "${cloneUrl}" ${destDir}`);
    cmd(`git branch -m ${destBranch}`, {cwd: destDir});
```

btw, good questions.  This is far from obvious.